lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -315,6 +315,10 @@ protected function _getCreateTableSQL($tableName, array $columns, array $options
      */
     public function getCreatePrimaryKeySQL(Index $index, $table)
     {
+        if ($table instanceof Table) {
+            $table = $table->getQuotedName($this);
```

Comment:
Should probably use a different variable name, so that we don't end up with reasoning about a union type at all